### DISCUSSION TITLE
Recommended setup for iBoost

### DISCUSSION THREAD

**Discussion by mudge** [springfall2008/batpred#1309] [Q&A]
Given the recent improvements to iBoost support in [8.3.0](https://github.com/springfall2008/batpred/releases/tag/v8.3.0) and my [previous question re iBoost mode](https://github.com/springfall2008/batpred/discussions/1258), what is the recommended setup for using predbat with one?

The [documentation mentions using a Fingerbot or similar device](https://springfall2008.github.io/batpred/customisation/#iboost-model-options) to physically press the "Boost" button but does that also mean somehow disabling the iBoost's automatic behaviour of diverting excess energy over its configurable export threshold? What are other iBoost users doing in this regard?

**Comment by Snid0** [comment]
I am also curious about how iboost works. I have an iboost (its actually a solaredge diverter). Not sure how this functionality works.

**Comment by mudge** [comment]
Has anyone successfully written an automation to trigger an iBoost with a FingerBot/SwitchBot when `binary_sensor.predbat_iboost_active` is `True` given that the manual boost option on the front of an iBoost only enables it for 15 minute increments up to 2 hours before cancelling the boost? Or is there some other recommended configuration of the iBoost and its sender and CT clamp to give predbat full control?

**Reply by armstrn2** [reply]
I wrote this in Home Assistant so that when the "Free Sessions" appear it turns the water on,     but there is NO WAY to stop the iBoost from running when it detects exporting !!!  -   which is why I just detect when it is hot 👍 

Here is my automation script 👍 ( your device ID's will be different 

alias: Solar Diverter - Free Energy Hours
description: Start/stop solar diverter based on Predbat determined slots
triggers:
  - entity_id:
      - predbat.rates
    trigger: state
actions:
  - choose:
      - conditions:
          - condition: numeric_state
            entity_id: predbat.rates
            below: 1
          - condition: template
            value_template: "{{ sensor.iboost_mode.state != \"Water Tank HOT\" }}"
        sequence:
          - device_id: f976c25a26ea549946a75649e24c6800
            domain: number
            entity_id: f47817041a56dc6309282f613f7c5cc5
            type: set_value
            value: 120
          - device_id: f976c25a26ea549946a75649e24c6800
            domain: button
            entity_id: ac26478a381fd8332914fc6917526b1a
            type: press
      - conditions:
          - condition: state
            entity_id: binary_sensor.predbat_iboost_active
            state: "False"
          - condition: numeric_state
            entity_id: predbat.rates
            above: 0
        sequence:
          - device_id: f976c25a26ea549946a75649e24c6800
            domain: button
            entity_id: 5f12a833bf12ee3c20cba58c78c7ceb4
            type: press
mode: single


![image](https://github.com/user-attachments/assets/568b35fb-c251-43e9-a291-249d96126f30)

![image](https://github.com/user-attachments/assets/7b8c4dd4-fd84-4866-be8c-af185daafb50)

**Reply by gcoan** [reply]
You should be able to get rid of the device id's and just use text entity id's which makes the script portable and more readable for others

**Reply by mudge** [reply]
How are you populating `sensor.iboost_mode.state`? Do you have a device reading the display on your iBoost?

The point re there being no way to stop the iBoost diverting energy when it detects export seems crucial: I wonder if predbat's iBoost functionality assumes it is really just controlling turning an electric immersion heater on and off (e.g. a literal switch) rather than accounting for iBoost's behaviour of turning on when an export threshold is met until the water is hot (or export dips back below the threshold).

**Reply by Ianwaltercoates** [reply]
I have been thinking about how to stop iBoost kicking in as well,  as you can't "technically" stop it when it detects excess power it will divert it,    So I was thinking of a sideways look and getting a solution in place that stops the sender from working !!

  So far the very simple thing would to replace the batteries in the sender with a FAKE USB one...   ( a fake AA battery that uses USB for power - i use them for xmas lights ),   and then have the USB charger plugged into a Smart plug,  so when I want to stop the iBOOST working I cut the power to the USB plug - and when no signal from the sender it stops all diversion ( boost still works though )

  But the only reason I can think of ever doing this is if I was going on holiday and didn't need FREE hot water whilst away - otherwise I'd not bother.

Now for the final bit....  I control my iBOOST+ using a handbuilt "Buddy" - https://github.com/JNSwanson/ESP-Home-iBoost/tree/main  this costs about £15 to make and integrates perfectly in Home Assistant,    so if you do have the newer iBoost and a soldering iron i'd recommend doing it.
